Copy link feature

elyessAR opened the pull request:
<!-- If your PR fixes an open issue, use `Closes #999` to link your PR with the issue. #999 stands for the issue number you are fixing -->

## Fixes Issue

closes #51 

## Changes proposed

<!-- List all the proposed changes in your PR -->

<!-- Mark all the applicable boxes. To mark the box as done follow the following conventions -->
<!--
[x] - Correct; marked as done
[X] - Correct; marked as done

[ ] - Not correct; marked as **not** done
-->

## Check List (Check all the applicable boxes) <!-- Follow the above conventions to check the box -->

- [ x] My code follows the code style of this project.
- [ ] My change requires changes to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] All new and existing tests passed.
- [x ] This PR does not contain plagiarized content.
- [ ] The title of my pull request is a short description of the requested changes.

## Screenshots

<!-- Add all the screenshots which support your changes -->

## Note to reviewers

<!-- Add notes to reviewers if applicable -->

furkanmutlu commented:
![image](https://user-images.githubusercontent.com/45897337/210002454-3d9a3e81-cf6b-43d7-b79e-e2f6ce220d0e.png)

hmm still looks like you have 3 commits. And also please close previous PR. Try to update your main branch with mine. Then create a branch, do your changes and push. 

Or you can reset back your changes on your current branch and rebase it to main. Then make your changes and push.

elyessAR commented:
![Screenshot](https://user-images.githubusercontent.com/49849558/210005150-d2de9041-0d81-4dd9-9357-2b6a569d4c57.png)
@furkanmutlu  I did exactly that, updated my main with yours, however the ahead commits didn't get removed, the only way I removed them is by creating a new branch from updated main  -> deleted the old one ->rename the new one to main however doing that require a new commit, I'm not sure but I guess my main will always be ahead a couple of commits even after rebase (because that's will require a new commit)  unless I refork and reclone the whole repo. correct me if I'm wrong

elyessAR commented:
> > I did exactly that, updated my main with yours, however the ahead commits didn't get removed, the only way I removed them is by creating a new branch from updated main -> deleted the old one ->rename the new one to main however doing that require a new commit, I'm not sure but I guess my main will always be ahead a couple of commits even after rebase (because that's will require a new commit) unless I refork and reclone the whole repo. correct me if I'm wrong
> 
> hmm.. First of all click Sync fork in your fork. Then go to your local main branch and do a `git pull`. Then go to your branch(copy-link-feature) try to do `git reset [commit id of the head]`. To see commit id for HEAD, do `git log` and copy id of the HEAd. Basically it will reset your commits to have a branch same with main. Then your commits will be converted to changes. Do `git stash` to stash your changes. Then just in case do `git rebase main`. Then bring your changes back with `git stash apply`. Then make your 1 commit. and force push `git push --force-lease`.

This worked! but only when I didn't rebase to main, I just did git reset to latest head commit, commited my changes and push.
I will try to do the same think to my main to fix it, Thank you!

furkanmutlu commented on frontend/src/components/Gallery/Overlay/Overlay.jsx:
```diff
@@ -32,6 +39,24 @@ const Overlay = ({
+        className="copy-button"
+        ref={target}
+        onClick={() => {
+          setShowToolTip(!showToolTip);
```
please move this function up

furkanmutlu commented on frontend/src/components/Gallery/Overlay/Overlay.jsx:
```diff
@@ -32,6 +39,24 @@ const Overlay = ({
+        onClick={() => {
+          setShowToolTip(!showToolTip);
+          onCopy();
+          setTimeout(function(){setShowToolTip(false);},2000);
```
```suggestion
          setTimeout(() => setShowTooltip(false), 2000);
```

furkanmutlu commented on frontend/src/components/Gallery/Overlay/Overlay.jsx:
```diff
@@ -32,6 +39,24 @@ const Overlay = ({
+      >
+        <FontAwesomeIcon icon={faLink} inverse/>
+      </button>
+        <CopyButtonOverlay target={target.current} show={showToolTip} placement="top" rootCloseEvent="mousedown">
```
Seems like there is spacing issue here, try to format the document.

furkanmutlu commented on frontend/src/components/Gallery/Overlay/Overlay.jsx:
```diff
@@ -32,6 +39,24 @@ const Overlay = ({
+        <FontAwesomeIcon icon={faLink} inverse/>
+      </button>
+        <CopyButtonOverlay target={target.current} show={showToolTip} placement="top" rootCloseEvent="mousedown">
+         {(props) => (
```
Do you need to pass this props ? Seems like there is no need for it.

furkanmutlu commented on frontend/src/components/Gallery/Overlay/Overlay.jsx:
```diff
@@ -1,16 +1,23 @@
+  onCopy,
 }) => {
+  const [showToolTip, setShowToolTip] = useState(false);
+  const target = useRef(null);
```
```suggestion
  const target = useRef();
```

furkanmutlu reviewed (approved):
Just added few changes. Seems like there is an issue with your linting. I also removed settiomeout to hide tooltip. Instead I added onmouse out event to do it.
